pkg/csi/csi_ops.go

```diff
@@ -187,14 +199,92 @@ func (c *applicationCreate) CreatePod(ctx context.Context, args *types.CreatePod
 	return podRes, nil
 }
 
+func (c *applicationCreate) WaitForPVCReady(ctx context.Context, namespace, name string) error {
+	if c.kubeCli == nil {
+		return fmt.Errorf("kubeCli not initialized")
+	}
+
+	err := c.waitForPVCReady(ctx, namespace, name)
+	if err != nil && (errors.Is(err, context.DeadlineExceeded) || strings.Contains(err.Error(), "would exceed context deadline")) {
+		eventErr := c.getErrorFromEvents(ctx, namespace, name, PVCKind)
+		if eventErr != nil {
+			return errors.Wrapf(eventErr, "had issues creating PVC")
+		}
+	}
+	return err
+}
+
+func (c *applicationCreate) waitForPVCReady(ctx context.Context, namespace string, name string) error {
+	pvcReadyTimeout := c.k8sObjectReadyTimeout
+	if pvcReadyTimeout == 0 {
+		pvcReadyTimeout = defaultReadyWaitTimeout
+	}
+
+	timeoutCtx, waitCancel := context.WithTimeout(ctx, pvcReadyTimeout)
+	defer waitCancel()
+	return poll.Wait(timeoutCtx, func(ctx context.Context) (bool, error) {
+		pvc, err := c.kubeCli.CoreV1().PersistentVolumeClaims(namespace).Get(timeoutCtx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, errors.Wrapf(err, "Could not find PVC")
+		}
+
+		if pvc.Status.Phase == v1.ClaimLost {
+			return false, fmt.Errorf("failed to create a PVC, ClaimLost")
+		}
+
+		return pvc.Status.Phase == v1.ClaimBound, nil
+	})
+}
+
 func (c *applicationCreate) WaitForPodReady(ctx context.Context, namespace string, podName string) error {
 	if c.kubeCli == nil {
 		return fmt.Errorf("kubeCli not initialized")
 	}
-	err := kankube.WaitForPodReady(ctx, c.kubeCli, namespace, podName)
+	err := c.waitForPodReady(ctx, namespace, podName)
+	if err != nil && (errors.Is(err, context.DeadlineExceeded) || strings.Contains(err.Error(), "would exceed context deadline")) {
```

Comment:
As above, can we just do `if err != nil` here? I don't see why we need to single out `context.DeadlineExceeded`, nor the string comparison.

Reply:
Yeah, or just wrap the warning events as string (if any) with the original event.